teamcity/resource_build_config.go

```diff
@@ -450,14 +450,41 @@ func resourceBuildConfigArchive(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
-	name := d.Get("name")
-	return client.BuildTypes.Rename(d.Id(), fmt.Sprintf("%s (Archived)", name.(string)))
+	parent, err := client.Projects.GetByID(d.Get("project_id").(string))
```

Comment:
I believe the use case that we need it for build configuration is:
1. Someone added a site
2. Archived it - It wasn't moved to Graveyard project
3. Added the site again